h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -114,25 +185,11 @@ public void initMetalearnerParams(Metalearner.Algorithm algo) {
    */
   @Override
   protected Frame predictScoreImpl(Frame fr, Frame adaptFrm, String destination_key, Job j, boolean computeMetrics, CFuncRef customMetricFunc) {
-    Frame levelOneFrame = new Frame(Key.<Frame>make("preds_levelone_" + this._key.toString() + fr._key));
-
-    // TODO: we should be able to parallelize scoring of base models
-    for (Key<Model> baseModelKey : _parms._base_models) {
-      if (isUsefulBaseModel(baseModelKey)) {
-        Model baseModel = baseModelKey.get();
-
-        Frame basePreds = baseModel.score(
-                fr,
-                "preds_base_" + this._key.toString() + fr._key,
-                j,
-                false
-        );
-
-        StackedEnsemble.addModelPredictionsToLevelOneFrame(baseModel, basePreds, levelOneFrame);
-        DKV.remove(basePreds._key); //Cleanup
-        Frame.deleteTempFrameAndItsNonSharedVecs(basePreds, levelOneFrame);
-      }
-    }
+    Key[] baseModels = Stream.of(_parms._base_models)
+            .filter(this::isUsefulBaseModel)
+            .toArray(Key[]::new);
+
+    Frame levelOneFrame = new StackedEnsembleMRScorer(fr, _key.toString(), j).doAll(baseModels).getLevelOneFrame();
```

Comment:
side note, wondering if we would profit from a similar approach when preparing levelOneFrame in `StackedEnsemble`, especially when using `blending` as in this case we need to explicitely score all base models against the blending frame.
Could be a future improvement.